ams/dist/src/main/arctic-bin/bin/optimizer.sh

```diff
@@ -37,14 +37,35 @@ else
 fi
 STDERR_LOG=${OPTIMIZER_LOG_DIR}/optimizer.log.err
 
-if [ ! -f "${OPTIMIZER_LOG_DIR}" ]; then
+if [ ! -d "${OPTIMIZER_LOG_DIR}" ]; then
   mkdir -p "${OPTIMIZER_LOG_DIR}"
 fi
 if [ ! -f $STDERR_LOG ];then
     touch $STDERR_LOG
 fi
 
-JAVA_OPTS="-Xmx$1m -Dlog.home=${OPTIMIZER_LOG_DIR}"
+JAVA_OPTS="-Xmx$2m -Dlog.home=${OPTIMIZER_LOG_DIR}"
 RUN_SERVER="com.netease.arctic.optimizer.standalone.StandaloneOptimizer"
 CMDS="$JAVA_RUN $JAVA_OPTS $RUN_SERVER $ARGS"
-nohup ${CMDS} >/dev/null 2>${STDERR_LOG} &
+
+function start {
```

Comment:
In sh, the defining function syntax is `function_name() { ... }`, while in bash, the commonly used syntax is function function_name { ... }, bash also supports the `function_name() { ... }` syntax, so it is recommended to use `function_name() { ... }`